ipl/main.c

```diff
@@ -665,31 +665,52 @@ void print_tsec_key()
 	{
 		EPRINTFARGS("Could not identify package1 version\nto read TSEC firmware (= '%s').",
 			(char *)pkg1 + 0x10);
+		
+		gfx_puts(&gfx_con, "\nPress any key...\n");
+		btn_wait();
 		goto out;
 	}
 
+	u8 keys[0x10 * 3];
 	for(u32 i = 1; i <= 3; i++)
 	{
-		u8 key[0x10];
-		int res = tsec_query(key, i, pkg1 + pkg1_id->tsec_off);
+		int res = tsec_query(keys + ((i - 1) * 0x10), i, pkg1 + pkg1_id->tsec_off);
 
 		gfx_printf(&gfx_con, "%kTSEC key %d: %k", 0xFF00DDFF, i, 0xFFCCCCCC);
 		if (res >= 0)
 		{
-			for (u32 i = 0; i < 0x10; i++)
-				gfx_printf(&gfx_con, "%02X", key[i]);
+			for (u32 j = 0; j < 0x10; j++)
+				gfx_printf(&gfx_con, "%02X", keys[((i - 1) * 0x10) + j]);
 		}
 		else
 			EPRINTFARGS("ERROR %X", res);
 		gfx_putc(&gfx_con, '\n');
 	}
+	
+		free(pkg1);
+	sdmmc_storage_end(&storage);
```

Comment:
Remove these 2.
They create undefined behavior when these 2 and the 2 at `out` label run.